scripts/run-coverage.sh

```
@@ -0,0 +1,81 @@
+BINARY="${1:-bin/moac-coverage}"
+if [ ! -f "$BINARY" ]; then
+    echo "Error: Coverage binary not found: $BINARY"
+    echo "Build it first with: make coverage"
+    echo ""
+    echo "Usage: $0 [binary] [args...]"
+    echo "Example: $0 bin/moac-coverage"
+    exit 1
+fi
+
+shift 2>/dev/null || true
+ARGS="$@"
```

Shellcheck (0.11.0)

[warning] Assigning an array to a string! Assign as array, or use * instead of @ to concatenate.

(SC2124)